packages/jupyter-ai/jupyter_ai/extension.py

```diff
@@ -302,20 +313,23 @@ async def get_chat(self, room_id: str) -> YChat:
         return document
 
     def on_change(self, room_id: str, events: ArrayEvent) -> None:
+        persona_manager = self.persona_managers_by_room.get(room_id, None)
         assert self.serverapp
+        assert persona_manager
 
         for change in events.delta:  # type:ignore[attr-defined]
             if not "insert" in change.keys():
                 continue
-            messages = change["insert"]
-            for message_dict in messages:
-                message = Message(**message_dict)
-                if message.sender == BOT["username"] or message.raw_time:
-                    continue
 
-                self.serverapp.io_loop.asyncio_loop.create_task(  # type:ignore[attr-defined]
-                    self.route_human_message(room_id, message)
-                )
+            # the "if not m['raw_time']" clause is necessary because every new
+            # message triggers 2 events, one with `raw_time` set to `True` and
+            # another with `raw_time` set to `False` milliseconds later.
+            # we should explore fixing this quirk in Jupyter Chat.
```

Comment:
consider adding a ref to an issue in `JupyterChat`

Reply:
Good callout. There wasn't an existing issue for this, so I opened a new one: https://github.com/jupyterlab/jupyter-chat/issues/212.

Added a reference as an inline comment too.